### PR TITLE
Switch from RFC3339 to ISO8601 for timestamp encoding

### PR DESCRIPTION
**Pull request by 0b11001111**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users.
---

Implement changes proposed in https://github.com/georust/gpx/issues/77

I don't expect this to land directly as it adds a breaking change

### CODE REVIEW

**Review by lnicola — APPROVED** [review]
This is technically breaking change, should probably bump the version number.

And please add an entry to the changelog.

**Comment by 0b11001111** [comment]
I added a few more changes to make my linter happy and thereby found a bug in the parsing logic for tracks.
On my machine, all unit and integrations tests pass :)

**Comment by 0b11001111** [comment]
I will later add another test case using the file exported from outdooractive.com that originally caused the RFC 3339 parsing error

File: src/parser/track.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ pub fn consume<R: Read>(context: &mut Context<R>) -> GpxResult<Track> {
             },
             XmlEvent::EndElement { ref name } => {
                 if name.local_name != "trk" {
-                    GpxError::InvalidClosingTag(name.local_name.clone(), "track");
+                    return Err(GpxError::InvalidClosingTag(name.local_name.clone(), "track"));
```
Comment (lnicola) [inline src/parser/track.rs:63]: :heart:

**Comment by lnicola** [comment]
Sorry for taking so long to look at this again, but can you replace the new test file with a smaller version (1-2 points or whatever)? I know that some of the other fixtures are already quite large, but it seems somewhat excessive.

**Comment by 0b11001111** [comment]
You're right, I filtered most waypoints out in https://github.com/georust/gpx/pull/78/commits/d3375a47a2eed0e582d52b152b8408a4366066cd

**Comment by lnicola** [comment]
Thanks. Can you also squash, since there's not much benefit in trimming down the file otherwise? :smile:

**Comment by 0b11001111** [comment]
I was today years old when I learned...

Hope I did the right thing :D

**Comment by lnicola** [comment]
Huh

bors ping

**Comment by lnicola** [comment]
Ah, this isn't your fault, we need to test with newer compiler versions.

**Comment by michaelkirk** [comment]
> Ah, this isn't your fault, we need to test with newer compiler versions.

I just now opened https://github.com/georust/gpx/pull/80 to address this.

**Comment by lnicola** [comment]
bors retry

**Comment by michaelkirk** [comment]
I took the liberty of fixing the conflicts in the change log.

bors retry

**Pull request merged**